test/Confluent.Kafka.IntegrationTests/Tests/Consumer_Assign.cs

```diff
@@ -0,0 +1,87 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+#pragma warning disable xUnit1026
+
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+
+namespace Confluent.Kafka.IntegrationTests
+{
+    public partial class Tests
+    {
+        /// <summary>
+        ///     Test <see cref="Consumer.Assign" /> with Offset.Beginning and Offset.End
+        /// </summary>
+        [Theory, MemberData(nameof(KafkaParameters))]
+        public void Consumer_Assign(string bootstrapServers)
+        {
+            LogToFile("start Consumer_Assign");
+
+            int N = 5;
+
+            Action<IConsumer<byte[], byte[]>> test = consumer => {
+                using (var topic = new TemporaryTopic(bootstrapServers, 1))
+                {
+                    Util.ProduceNullStringMessages(bootstrapServers, topic.Name, 1, N);
+                    consumer.Assign(new TopicPartitionOffset(topic.Name, 0, Offset.Beginning));
+                    var cr1 = consumer.Consume(TimeSpan.FromSeconds(10));
+                    Assert.Equal(0, cr1.Offset);
+                    // test that if consumer is paused and reassigned, this does not affect offset.
+                    consumer.Pause(new List<TopicPartition> { new TopicPartition(topic.Name, 0) });
+                    consumer.Assign(new TopicPartitionOffset(topic.Name, 0, Offset.End));
+                    var cr2 = consumer.Consume(1000);
+                    Assert.Null(cr2);
+                    Util.ProduceNullStringMessages(bootstrapServers, topic.Name, 1, N);
+                    consumer.Resume(new List<TopicPartition> { new TopicPartition(topic.Name, 0) });
+                    var cr3 = consumer.Consume(TimeSpan.FromSeconds(10));
+                    Assert.Equal(5, cr3.Offset);
+                    var p = consumer.Position(new TopicPartition(topic.Name, 0));
+                    Assert.Equal(6, p.Value);
+                    consumer.Assign(new TopicPartitionOffset(topic.Name, 0, Offset.Beginning));
+                    var cr4 = consumer.Consume(TimeSpan.FromSeconds(10));
+                    Assert.Equal(0, cr4.Offset);
+                }
+            };
+
+            var consumerConfig = new ConsumerConfig
+            {
+                GroupId = Guid.NewGuid().ToString(),
+                BootstrapServers = bootstrapServers,
+                SessionTimeoutMs = 6000,
+                EnableAutoCommit = false,
+                AutoOffsetReset = AutoOffsetReset.Earliest
```

Comment:
Set this to .Error to make sure the Assign():ed offset is used and not auto.offset.reset.

Reply:
don't you think it's a better test as is? (since the behavior if `Earliest` should still be to honor the offset passed into `Assign`, and there is probably more chance the code doesn't work this way than with `Error`).

Reply:
Since you are passing Beginning (same as Earliest) to assign there is no way to distinguish if it was the offset passed to Assign() or auto.offset.reset that was used.
Setting it to Error will raise an error if the auto.offset.reset is used, which is a stricter (and thus better) test case.

Reply:
ahh right, i see your point now.